### ISSUE TITLE
模型超过2G怎么处理

### ISSUE THREAD

**Issue by Worromots** [ModelTC/Dipoorlet#32]
File "/workdir/common/anaconda3/envs/qpixart/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/mnt/dolphinfs/hdd_pool/docker/user/hadoop-platcv/tsl/q_mtpixart/Dipoorlet/dipoorlet/__main__.py", line 120, in <module>
    act_clip_val, weight_clip_val = tensor_calibration(onnx_graph, args)
  File "/mnt/dolphinfs/hdd_pool/docker/user/hadoop-platcv/tsl/q_mtpixart/Dipoorlet/dipoorlet/tensor_cali/tensor_cali_base.py", line 6, in tensor_calibration
    act_clip_val = tensor_cali_dispatcher(args.act_quant, onnx_graph, args)
  File "/mnt/dolphinfs/hdd_pool/docker/user/hadoop-platcv/tsl/q_mtpixart/Dipoorlet/dipoorlet/utils.py", line 297, in wrapper
    return dispatch(args[0])(*(args[1:]), **kw)
  File "/mnt/dolphinfs/hdd_pool/docker/user/hadoop-platcv/tsl/q_mtpixart/Dipoorlet/dipoorlet/tensor_cali/basic_algorithm.py", line 35, in find_clip_val_hist
    stats_min_max = forward_get_minmax(onnx_graph, args)
  File "/mnt/dolphinfs/hdd_pool/docker/user/hadoop-platcv/tsl/q_mtpixart/Dipoorlet/dipoorlet/forward_net.py", line 200, in forward_get_minmax
    ort_session = ort.InferenceSession(net.SerializeToString(), providers=providers)

**Comment by gushiqiao** [comment]
跑的是transformer类的模型嘛？

**Comment by Worromots** [comment]
是的 跑的DIT发自我的 iPhone在 2024年8月26日，14:51，gushiqiao ***@***.***> 写道：﻿
跑的是transformer类的模型嘛？

**Comment by gushiqiao** [comment]
超过2G的模型，Dipoorlet会依赖"onnxruntime.transformers.optimizer"进行优化， 具体代码在这里，https://github.com/ModelTC/Dipoorlet/blob/c89130744d45ae2e7cb77081b8799c2ff31ee08d/dipoorlet/__main__.py#L85。
至于DIT，其实要看"onnxruntime.transformers.optimizer"是否支持，目前我们在Dipoorlet里只支持了Unet的优化，可以在这里看到https://github.com/ModelTC/Dipoorlet/blob/c89130744d45ae2e7cb77081b8799c2ff31ee08d/dipoorlet/__main__.py#L53

**Comment by gushiqiao** [comment]
> 超过2G的模型，Dipoorlet会依赖"onnxruntime.transformers.optimizer"进行优化， 具体代码在这里，[https://github.com/ModelTC/Dipoorlet/blob/c89130744d45ae2e7cb77081b8799c2ff31ee08d/dipoorlet/__main__.py#L85。](https://github.com/ModelTC/Dipoorlet/blob/c89130744d45ae2e7cb77081b8799c2ff31ee08d/dipoorlet/__main__.py#L85%E3%80%82) 至于DIT，其实要看"onnxruntime.transformers.optimizer"是否支持，目前我们在Dipoorlet里只支持了Unet的优化，可以在这里看到
> 
> https://github.com/ModelTC/Dipoorlet/blob/c89130744d45ae2e7cb77081b8799c2ff31ee08d/dipoorlet/__main__.py#L53

@Worromots
